From d8dda221feb33c9e75c7c86414597aa36c577471 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sun, 16 Jul 2017 20:40:26 -0400
Subject: [PATCH] SPIGOT-3427: Recalculate ConsoleSender permissions after
 plugins have loaded

Resolves a really weird race condition where the console sender could lose permission to execute
commands defined in plugin.yml with default=op

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 9ef2e4a55..938602af1 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -340,6 +340,7 @@ public final class CraftServer implements Server {
             if (!com.destroystokyo.paper.PaperConfig.loadPermsBeforePlugins) loadCustomPermissions(); // Paper
             DefaultPermissions.registerCorePermissions();
             CraftDefaultPermissions.registerCorePermissions();
+            getConsoleSender().recalculatePermissions(); // Paper - SPIGOT-3427
             helpMap.initializeCommands();
         }
     }
-- 
2.13.0

